Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHand.cs

```diff
@@ -117,6 +118,7 @@ public void UpdateState(SimulatedHandData handData)
                 }
                 else
                 {
+                    jointPositions[i] = Vector3.SmoothDamp(jointPoses[handJoint].Position, jointPositions[i], ref jointPositionVelocities[i], 0.15f);
```

Comment:
I would like to restrict the smoothing to just the mouse movement, which seems to be it's intended purpose. The gesture animation of the actual joints is already animated smoothly, so the damping just makes it feel too "squishy" IMO and isn't necessary to avoid fast motion.

Doing this in the `SimulatedHandDataProvider` instead has the added advantage that smoothing is not applied when hands are simply animated. This is important for the upcoming input recording feature, which will be used for testing and should reproduce the recorded animation as accurately as possible, without additional smoothing.

```diff
diff --git a/Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs b/Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs
index d348e3dec..53b355145 100644
--- a/Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs
+++ b/Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedHandDataProvider.cs
@@ -31,6 +31,7 @@ internal class SimulatedHandState
         public bool IsPinching { get; private set; }

         public Vector3 ScreenPosition;
+        private Vector3 velocity = Vector3.zero;
         // Rotation of the hand
         public Vector3 HandRotateEulerAngles = Vector3.zero;
         // Random offset to simulate tracking inaccuracy
@@ -90,17 +91,21 @@ public void SetViewportPosition(Vector3 point)
             ScreenPosition = CameraCache.Main.ViewportToScreenPoint(point);
         }

-        public void SimulateInput(Vector3 mouseDelta, float noiseAmount, Vector3 rotationDeltaEulerAngles)
+        public void SimulateInput(Vector3 mouseDelta, float noiseAmount, Vector3 rotationDeltaEulerAngles, float smoothTime)
         {
+            Vector3 worldPoint = CameraCache.Main.ScreenToWorldPoint(ScreenPosition);
+            Vector3 newScreenPosition = ScreenPosition;
             // Apply mouse delta x/y in screen space, but depth offset in world space
-            ScreenPosition.x += mouseDelta.x;
-            ScreenPosition.y += mouseDelta.y;
-            Vector3 newWorldPoint = CameraCache.Main.ScreenToWorldPoint(ScreenPosition);
+            newScreenPosition.x += mouseDelta.x;
+            newScreenPosition.y += mouseDelta.y;
+            Vector3 newWorldPoint = CameraCache.Main.ScreenToWorldPoint(newScreenPosition);
             newWorldPoint += CameraCache.Main.transform.forward * mouseDelta.z;
-            ScreenPosition = CameraCache.Main.WorldToScreenPoint(newWorldPoint);

-            HandRotateEulerAngles += rotationDeltaEulerAngles;
+            // Apply damping to the hand movement
+            newWorldPoint = Vector3.SmoothDamp(worldPoint, newWorldPoint, ref velocity, smoothTime);

+            ScreenPosition = CameraCache.Main.WorldToScreenPoint(newWorldPoint);
+            HandRotateEulerAngles += rotationDeltaEulerAngles;
             JitterOffset = UnityEngine.Random.insideUnitSphere * noiseAmount;
         }

@@ -295,7 +300,7 @@ private void SimulateUserInput()

             if (isSimulating)
             {
-                state.SimulateInput(mouseDelta, profile.HandJitterAmount, rotationDeltaEulerAngles);
+                state.SimulateInput(mouseDelta, profile.HandJitterAmount, rotationDeltaEulerAngles, profile.HandSmoothTime);

                 if (isAlwaysVisible)
                 {

```